File: lib/models/business.dart
class Business {
  final int id;
  final String name;
  final String llcName;
  final String address;
  final String? logoUrl;
  final String timezone;
  final String autoGratuityPercent;
  final int autoGratuityMinGuests;
  final String ccFeePercent;
  final DateTime createdAt;
  final DateTime updatedAt;

  Business({
    required this.id,
    required this.name,
    required this.llcName,
    required this.address,
    this.logoUrl,
    required this.timezone,
    required this.autoGratuityPercent,
    required this.autoGratuityMinGuests,
    required this.ccFeePercent,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Business.fromJson(Map<String, dynamic> json) {
    return Business(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      llcName: json['llc_name'] ?? '',
      address: json['address'] ?? '',
      logoUrl: json['logo_url'],
      timezone: json['timezone'] ?? 'UTC',
      autoGratuityPercent: json['auto_gratuity_percent'] ?? '0.00',
      autoGratuityMinGuests: json['auto_gratuity_min_guests'] ?? 0,
      ccFeePercent: json['cc_fee_percent'] ?? '0.00',
      createdAt: DateTime.parse(json['created_at'] ?? DateTime.now().toIso8601String()),
      updatedAt: DateTime.parse(json['updated_at'] ?? DateTime.now().toIso8601String()),
    );
  }
}

File: app/Models/Business.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Business extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'llc_name',
        'address',
        'logo_url',
        'timezone',
        'auto_gratuity_percent',
        'auto_gratuity_min_guests',
        'cc_fee_percent',
    ];

    protected $casts = [
        'auto_gratuity_percent' => 'decimal:2',
        'cc_fee_percent' => 'decimal:2',
    ];

    public function employees(): HasMany
    {
        return $this->hasMany(Employee::class);
    }

    public function users(): HasMany
    {
        return $this->hasMany(User::class);
    }

    public function roles(): HasMany
    {
        return $this->hasMany(Role::class);
    }

    public function discountReasons(): HasMany
    {
        return $this->hasMany(DiscountReason::class);
    }

    public function floors(): HasMany
    {
        return $this->hasMany(Floor::class);
    }

    public function menus(): HasMany
    {
        return $this->hasMany(Menu::class);
    }

    public function menuCategories(): HasMany
    {
        return $this->hasMany(MenuCategory::class);
    }

    public function menuItems(): HasMany
    {
        return $this->hasMany(MenuItem::class);
    }

    public function printers(): HasMany
    {
        return $this->hasMany(Printer::class);
    }

    public function printerRoutes(): HasMany
    {
        return $this->hasMany(PrinterRoute::class);
    }

    public function taxRates(): HasMany
    {
        return $this->hasMany(TaxRate::class);
    }

    public function reservations(): HasMany
    {
        return $this->hasMany(Reservation::class);
    }

    public function gratuitySetting(): HasOne
    {
        return $this->hasOne(GratuitySetting::class);
    }
}
